js/bitfinex2.js

```diff
@@ -1440,8 +1440,15 @@ module.exports = class bitfinex2 extends bitfinex {
             }
             tag = this.safeString (data, 3);
             type = 'withdrawal';
-        } else {
+        } else if (transactionLength === 22) {
             id = this.safeString (transaction, 0);
+            if (currency !== undefined) {
+                code = currency['code'];
+            } else {
+                const currencyId = this.safeString (transaction, 1);
+                currency = this.safeCurrency (currencyId, currency);
+                code = currency['code'];
+            }
```

Comment:
Let's rewrite the above lines 1445-1451, since `if (currency !== undefined)` is already done in `safeCurrency`:

```JavaScript
const currencyId = this.safeString (transaction, 1);
currency = this.safeCurrency (currencyId, currency);
code = currency['code'];
```